EFRS-1333: Added the ability to send embeddings instead of the image for recognition (#948)
exadel-inc/CompreFace: docs/Rest-API-description.md

-d {"source_image": "<source_image_base64_value>", "target_image": "<target_image_base64_value>"}
```

> ## Embedding Support

Reviewer: ```suggestion
## Recognition and verification using embedding
```

`since 1.2.0 version`

If you already have a computed embedding, you can use it to perform recognition and verification.
The base rule is to use `Content-Type: application/json` header and send JSON in the body.

> ### Embedding Recognition Service

Reviewer: ```suggestion
### Recognize Faces from a Given Image, Embedding
```

> The service is used to determine similarities between input embeddings and embeddings within the database. An example:

Reviewer: ```suggestion
The service is used to determine similarities between input embeddings and embeddings within the Face Collection. An example:
```

Response body on success:
```json
{
  "results": [
    {
      "embedding": [0.0627421774604647, "...", -0.0236684433507126],
      "similarities": [
        {
          "subject": "John",
          "similarity": 0.55988
        },
        "..."
      ]
    },
    "..."
  ]
}
```

| Element      | Type   | Description                                                                                |
|--------------|--------|--------------------------------------------------------------------------------------------|
| results      | array  | an array that contains all the results                                                     |
| embedding    | array  | an embedding that is similar to the input embedding                                        |
| similarities | array  | an array that contains results of similarity between the embedding and the input embedding |
| subject      | string | a subject in which the similar embedding was found                                         |
| similarity   | float  | a similarity between the embedding and the input embedding                                 |

> ### Embedding Verification Service

Reviewer: ```suggestion
### Face Verification Service, Embedding
```

| Content-Type     | header      | string  | required | application/json                                                   |
| x-api-key        | header      | string  | required | api key of the Embedding verification service, created by the user |
| source           | body        | array   | required | the source embedding. The length must be 512                       |
| targets          | body        | array   | required | the target embeddings. The length must be 512 for each of them     |

Response body on success:
```json
{
  "results": [
    {
      "embedding": [0.0627421774604647, "...", -0.0236684433507126],
      "similarity": 0.55988
    },
    "..."
  ]
}
```

| Element      | Type   | Description                                                        |
|--------------|--------|--------------------------------------------------------------------|
| results      | array  | an array that contains all the results                             |
| embedding    | array  | a target embedding that is similar to the source embedding         |
| similarity   | float  | a similarity between the source embedding and the target embedding |

> ### Verify Embeddings using Given Embedding

Reviewer: ```suggestion
### Verify Faces from a Given Image, Embedding
```

      ]
    },
    "..."
  ]
}
```

| Element      | Type   | Description                                                                                |
|--------------|--------|--------------------------------------------------------------------------------------------|
| result       | array  | an array that contains all the results                                                     |
| embedding    | array  | an input embedding                                                                         |
| similarities | array  | an array that contains results of similarity between the embedding and the input embedding |
| subject      | string | a subject in which the similar embedding was found                                         |
| similarity   | float  | a similarity between the embedding and the input embedding                                 |

### Face Verification Service, Embedding
The service is used to determine similarities between an input source embedding and input target embeddings. An example:

```shell
curl -X POST "http://localhost:8000/api/v1/verification/embeddings/verify" \
-H "Content-Type: application/json" \
-H "x-api-key: <service_api_key>" \

> -d {"source": [<source_embedding>], "targets": [["target_embedding"], ...]}

Reviewer: ```suggestion
-d {"source": [<source_embedding>], "targets": [[<target_embedding>], ...]}
```

| x-api-key        | header      | string  | required | api key of the Face verification service, created by the user                        |
| source           | body        | array   | required | an input embeddings. The length depends on the model (e.g. 512 or 128)               |
| targets          | body        | array   | required | an array of the target embeddings. The length depends on the model (e.g. 512 or 128) |

Response body on success:
```json
{
  "result": [
    {
      "embedding": [0.0627421774604647, "...", -0.0236684433507126],
      "similarity": 0.55988
    },
    "..."
  ]
}
```

| Element     | Type   | Description                                                        |
|-------------|--------|--------------------------------------------------------------------|
| result      | array  | an array that contains all the results                             |
| embedding   | array  | a target embedding which we are comparing to source embedding      |
| similarity  | float  | a similarity between the source embedding and the target embedding |

> ### Verify Faces from a Given Image, Embedding

Reviewer: Make the order of services:
Recognize Faces from a Given Image, Embedding
Verify Faces from a Given Image, Embedding
Face Verification Service, Embedding
